webhook_to_fedora_messaging/endpoints/parser/parser.py

```diff
@@ -0,0 +1,63 @@
+
+from  flask import Request
+from werkzeug.datastructures.headers import EnvironHeaders
+import hashlib
+import hmac
+from webhook_to_fedora_messaging_messages.github.github import GithubMessageV1
+from ..util import fas_by_github
+
+
+
+def msg_parser(service_type: str, request: Request):
+    if service_type.lower() == "github":
+        return _github_parser(request)
+
+
+def _github_parser(request: Request, secret=True) -> GithubMessageV1:
+    """
+        Convert Flask request objects into desired FedMsg format.
+        
+        Args:
+            headers: Flask request.headers
+            body: A dictionary contains payload
+            secret: Specifies whether the webhook has secret key feature on or not
+    """
+    
+    headers = get_github_headers(request.headers)
+    
+    secret = "00b863c62c96aa67aeea966104362b1136ce883d" # will come from a data source
+    if (secret and verify_signature(request.data, secret, headers['X-Hub-Signature-256'])) or not secret:
+        topic = "github." + headers['X-Github-Event']
+        agent =  fas_by_github(request.json['sender']['login']) # FASJSON
+        return GithubMessageV1(topic=topic, body={'body': request.json, 'headers': headers, 'agent': agent})
+    return None
```

Comment:
If the signature didn't match, we should raise an exception. A custom exception (to put in `webhook_to_fedora_messaging.exceptions`) is best, but a ValueError could also do.